hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSourceWALReader.java

```diff
@@ -409,17 +413,7 @@ private boolean acquireBufferQuota(long size) {
     return totalBufferUsed.addAndGet(size) >= totalBufferQuota;
   }
 
-  /**
-   * @return whether the reader thread is running
-   */
-  public boolean isReaderRunning() {
-    return isReaderRunning && !isInterrupted();
```

Comment:
Shouldn't it also be looking for an interrupt? (like in a  thread pool shutdown).